manifests/packages.pp

```diff
@@ -78,11 +80,26 @@
             ensure  => $docker_version,
             require => Class['Apt::Update'],
           }
-        }
-        else {
+          if $pin_packages {
+            file { '/etc/apt/preferences.d/docker':
+              mode    => '0444',
+              owner   => 'root',
+              group   => 'root',
+              content => template('kubernetes/docker_apt_package_pins.erb'),
+              notify  => Service['docker'],
+            }
+          }else {
+            file { '/etc/apt/preferences.d/docker':
+              ensure => absent,
+            }
+          }
+        }else {
           package { $docker_package_name:
             ensure => $docker_version,
           }
+          if $pin_packages {
```

Comment:
Can you help me understand this condition - if someone has **not** set the `create_repos` and `manage_docker` parameters, indicating that they do not want to install and manage Docker, would we even need to fail if `pin_packages` is set? 

I guess it is good to warn and indicate that the configuration you set is not going to be taken in to consideration - would that suffice or is there something I'm missing?

Reply:
Your are right, safety was the intention of the fail() message.
Accidentally i missed to describe this in the fail() message - i improved that now.